### ISSUE TITLE
linter: `prefer-destructuring` doesn't consider type annotations

### ISSUE THREAD

**Issue by elusiveunit** [oxc-project/oxc#24615]
### What version of Oxlint are you using?

1.74.0

### What command did you run?

npx oxlint

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```typescript
export default defineConfig({
	rules: {
		'prefer-destructuring': [
			'warn',
			{
				VariableDeclarator: { array: false, object: true },
				AssignmentExpression: { array: false, object: false },
			},
			{ enforceForRenamedProperties: false },
		],
	},
});
```

### What happened?

Same issue from typescript-eslint: https://github.com/typescript-eslint/typescript-eslint/issues/723

example.ts
```typescript
const foo = { bar: 1 as const };
const bar: number = foo.bar;
console.log(bar);
```
Output from `npx oxlint`:
```
  ⚠ eslint(prefer-destructuring): Use Object destructuring.
   ╭─[example.ts:2:21]
 1 │ const foo = { bar: 1 as const };
 2 │ const bar: number = foo.bar;
   ·                     ───────
 3 │ console.log(bar);
   ╰────
  help: Use object destructuring rather than direct member access.
```
The auto fix suggests `const {bar} = foo;` but that changes the `bar` type. The equivalent ESLint config with `@typescript-eslint/prefer-destructuring` only reports if the `number` annotation is removed.
